src/roomba_robot/tasks/botgal_and_lib/botgal_task.c

```diff
@@ -0,0 +1,190 @@
+    printf("%d,%d,%s", get_create_distance(), 0, "total distance\n");
+    printf("%d,%d,%s", get_create_total_angle(), 0, "total angle \n");
+}
+
+bool setupcamera() {
+    if (camera_open() == 0 || camera_update() == 0) return false;
+    return true;
+}
+
+void setuprobot(bool debug, bool camera) {
+    create_connect();
+    setupsensordata();
+#ifdef DEBUG
+    printsensordata();
+#endif
+    if (camera) {
+        if (setupcamera()) printf("%s", "Camera initialized");
+        else printf("%s", "Camera Error");
+    }
+    enable_servos();
+}
+
+void shutdownrobot(bool camera) {
+    if (camera) {
+        camera_close();
+    }
+    disable_servos();
+    ao();
+    create_stop();
+    create_disconnect();
+}
+
+void drivestraight(int speed, enum direction{forward,backward}; direction, int distance,bool watch_for_touch) { // distance in mm
+    if (direction == forward) {
+        set_create_distance(0);
+        while (get_create_distance() <= (distance)) {
+            create_drive_direct(speed, speed);
+            if (watch_for_touch) {
+                if (get_create_lbump() || get_create_rbump() == true) {
+                    set_create_distance(distance);
+                }
+            }
+        }
+    } else {
+        set_create_distance(0);
+        while ((-1 * get_create_distance()) <= (distance)) {
+            create_drive_direct(speed * -1, speed * -1);
+        }
+    }
+}
+
+void turn(int speed, int deg) {
+    // deg - turn to right  + turn to left
+    set_create_distance(0);
+    set_create_total_angle(0);
+    if (deg > 0) {
+        while (get_create_total_angle() <= deg) {
+            create_drive_direct(speed * -1, speed);
+        }
+    } else {
+        while (get_create_total_angle() >= deg) {
+            create_drive_direct(speed, speed * -1);
+        }
+    }
+}
+
+void wasd() { // function to control robot via webinterface
+    bool exit;
+    exit = true;
+    int speed = 100;
+    while (exit) {
+        char input;
+        printf("%s", "Please provide input:\n");
+        scanf("%c", &input);
+        switch (input) {
+            case 'w':
+                drivestraight(speed, forward, 25, true);
+                break;
+            case 's':
+                drivestraight(speed, backward, 25, true);
+                break;
+            case 'd':
+                turn(speed / 2, -45);
+                break;
+            case 'a':
+                turn(speed / 2, 45);
+                break;
+            case 'x':
+                exit = false;
+                break;
+            case 'm':
+                printf("%s", "Please provide speed:\n");
+                scanf("%d", &speed);
+                break;
+            default:
+                printf("%s", "Invalid input\n");
+                break;
+        }
+    }
+}
+
+void calibration(){
+    // default positions
+    int servo_arm = 2000;
+    int servoTurnArm = 1500;
+    int servoHand = 100;
+    int motorArm = 12700;
```

Comment:
I thought that it was obvious with just one example, sorry:

I meant all of these variables do not have the same code standard:

```c
int servo_arm = 2000;
int servo_turn_arm = 1500;
int servo_hand = 100;
int motor_arm = 12700;
```